### PR TITLE
[English readme][bsp][stm32] add nucleo-144 doc

### PR DESCRIPTION
**Pull request by mysterywolf**
## 拉取/合并请求描述：(PR description)

[
add an English document for STM32 NUCLEO-144 boards.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+- Board-specific features
+  - External or internal SMPS to generate Vcore logic supply
+  - Ethernet compliant with IEEE-802.3-2002
+  - USB OTG full speed or device only
+  - Board connectors: 
+    - USB with Micro-AB or USB Type-C™ 
+    - Ethernet RJ45
+  - Arm® Mbed Enabled™ compliant
+
+### **For more details about these boards, please refer to [ST Nucleo Official Website](https://www.st.com/en/evaluation-tools/stm32-nucleo-boards.html?querycriteria=productId=LN1847).**
```
Comment (Cathy-lulu) [inline bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md:47]: For more details about this board.

File: bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+> https://www.youtube.com/watch?v=ab4ilbsteWU
+>
+> https://www.youtube.com/watch?v=dO-BMOzNKcI
+
+
+
+## Advanced Features
+
+This BSP only enables GPIO and USART2 by default. If you need more advanced features such as SPI, I2C, you need to configure the BSP with RT-Thread [ENV tool](https://www.rt-thread.io/download.html?download=Env) , as follows:
```
Comment (Cathy-lulu) [inline bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md:105]: Note that "nv" of "Env" should take the lower case as Env is a branch product of RT-Thread when we are writing or publishing it we should keep it the same as the RT-Thread official publishing.

File: bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+1. Open the ENV tool under the specific BSP folder;
+2. Enter `menuconfig` command to configure the project, then save and exit;
+3. Enter `pkgs --update` command to update the package;
+4. Enter `scons --target=mdk4/mdk5/iar` command to regenerate the project.
+
+Learn how to use RT-Thread ENV, click [Here](https://github.com/RT-Thread/rtthread-manual-doc/blob/master/env/env.md).
```
Comment (Cathy-lulu) [inline bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md:112]: Note that "nv" of "Env" should take the lower case as Env is a branch product of RT-Thread when we are writing or publishing it we should keep it the same as the RT-Thread official publishing.

File: bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md (hunk 1 above)
Comment (mysterywolf) [reply bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md:47]: Negative, "Nucleo-144" is a series of boards and has many different sub-types.

Comment (Cathy-lulu) [reply bsp/stm32/docs/STM32_Nucleo-144_BSP_Introduction.md:47]: Ohhh, I see then. Thanks for sharing.

**Review by Cathy-lulu — APPROVED** [review]

**Pull request merged**